src/app.js

```diff
@@ -1,8 +1,34 @@
 import config from './config';
 import Api from './Api';
+import InfoBarWidget from './InfoBarWidget';
+let components = $( 'body' ).data( 'wwt-components' );
 
-// eslint-disable-next-line no-unused-vars
-const a = new Api( { url: config.wikicolorUrl } ); // Test
-// TEST!
+// Initialize
+if ( !components ) {
+	// Store, so we don't append and create objects if they've been created
+	// TODO: We could potentially store all that data in some view model that's shared
+	// across the widgets.
+	components = {
+		widget: new InfoBarWidget(),
+		api: new Api( { url: config.wikicolorUrl } ),
+		$originalOutput: $( 'body' ).data( 'wwt-originalOutput' )
+	};
 
-OO.ui.alert( 'The extension is working! URL: ' + a.getAjaxURL( window.location.href ) );
+	$( 'body' ).data( 'wwt-components', components );
+
+	if ( $( 'body' ).hasClass( 'skin-vector' ) ) {
```

Comment:
`$( '#mw-head-base' ).after( components.widget.$element )` should work in all skins. If not, could I beg to also put `|| $( 'body' ).hasClass( 'timeless' )`? :)

Reply:
Unfortunately, I just checked in the console:

`$( '#mw-head-base' )` exists in Vector.
`$( '#mw-head-base' )` does not exist in Timeless.
`$( '#mw-head-base' )` does not exist in Monobook.

I'll put in a correction for Timeless, and even if we find a consistent placement, we will still need some corrections per skin, probably, because the widths of the elements are not consistent, and the bar needs to fit the width of the content (and the head is outside it)

With my testing, it seemed most straight forward (with the least per-skin CSS corrections) to find a different placement for each of those. We can iterate later, but I'll at least make sure that in this patch it works without being too mangled.

Reply:
I could have sworn I tested this! Apparently not. We could put the bar before `#firstHeading` but this would compromise the design due to the padding around the outermost `.mw-body`.

Reply:
Yeah this isn't awesome :\

Okay, I went over all three skins and adjusted. I had to attach the widget a bit differently in each one because of visual considerations. In Timeless, for example, you already have a floating 'sticky' header, so the widget needed to be below it but constrained to the content area. In monobook, the spacing was a bit off, so I corrected with negative margins.

I also had to pull in the representation of the "close" icon into data-uri for Timeless and Monobook; it looks like Apex (ooui theme used in those two skins) doesn't have that icon. Bah.